### DISCUSSION TITLE
An indicator for unwatched aired episodes

### DISCUSSION THREAD

**Discussion by Mythreal93** [Goldenfreddy0703/Otaku#328] [Ideas]
Would be great if we could have some sort of indication or icon to show which anime in our lists have aired episodes we haven't seen yet.

It could be implemented via a dot in the Anime's title possibly?

**Comment by Goldenfreddy0703** [comment]
Let me guess, your the guy Schwaxx on reddit that made a post yesterday about episode indicators. Is that correct?

https://www.reddit.com/r/Addons4Kodi/s/nw26fbrxJM

If not then sorry about that and for episode indicators, I'm not sure if that is possible at the moment due to our watchlist cause we use simkl for episode meta and anilist and simkl episode counts can be different which is why, we do have completed indicators, just not watched indicators.

**Reply by Mythreal93** [reply]
I had actually looked into this a bit myself to see if I could figure it out and managed to piece together a method, but couldn't find a good place to implement it. Kodi stores your "watched" videos in a local database, you can use sqlite to grab the number of watched videos there. 

```
def get_latest_video_db_path():
    database_path = xbmcvfs.translatePath('special://database/')
    files = os.listdir(database_path)
    video_dbs = sorted([f for f in files if f.startswith('MyVideos') and f.endswith('.db')])
    if video_dbs:
        return os.path.join(database_path, video_dbs[-1])
    return None

def get_watched_status(filename):
    db_path = get_latest_video_db_path()
    if db_path is None:
        xbmc.log("No video database found.", xbmc.LOGERROR)
        return False
        
    try:
        connection = sqlite3.connect(db_path)
        cursor = connection.cursor()
        cursor.execute("SELECT playCount FROM files WHERE strFilename=?", (filename,))
        result = cursor.fetchone()
        connection.close()
        return result[0] > 0 if result else False
    except Exception as e:
        xbmc.log("Failed to query the database: " + str(e), xbmc.LOGERROR)
        return False
```

I know this isn't the correct place to implement it, but I was doing some testing and managed to get it partly working by testing it here:

```
@route('watchlist_to_ep/*')
def WATCHLIST_TO_EP(payload, params):
    payload_list = payload.rsplit("/")
    anilist_id, mal_id, kitsu_id, eps_watched = payload_list

    show = database.get_show_mal(mal_id) if mal_id else database.get_show(anilist_id)
    anilist_id = show['anilist_id']
    kodi_meta = pickle.loads(show['kodi_meta'])
    kodi_meta['eps_watched'] = eps_watched
    database.update_kodi_meta(anilist_id, kodi_meta)

    anime_general, content_type = OtakuBrowser().get_anime_init(anilist_id)
    filtered_anime_general = []

    if anime_general:
        for episode in anime_general:
            aired_info = episode.get('info', {}).get('aired')
            if aired_info and time.strptime(aired_info, '%Y-%m-%d') < time.localtime():
                if not get_watched_status(episode.get('url')):
                    filtered_anime_general.append(episode)
                    xbmc.log(f"Unwatched aired episode found at URL {episode.get('url')}", xbmc.LOGINFO)
                else:
                    xbmc.log(f"Episode at URL {episode.get('url')} is already watched", xbmc.LOGDEBUG)
    
    return control.draw_items(filtered_anime_general, content_type)
```

Maybe you're able to find a spot to get it working, or suggest a spot that this would be more appropriate so I can give it a whirl?

Thanks!

**Comment by Goldenfreddy0703** [comment]
Hey so i don't remember but i think we completed this. Hope this helps, Thanks
